From d794a84a9e147498069b4c582d57b899bf068187 Mon Sep 17 00:00:00 2001
From: "Andrew W. Nosenko" <andrew.w.nosenko@gmail.com>
Date: Mon, 15 Nov 2010 13:00:29 +0100
Subject: [PATCH] 599241 fix an initialization problem on Win64

as the thread_t is not an integral datatype on that platform
---
 testThreads.c |    6 ++----
 1 files changed, 2 insertions(+), 4 deletions(-)

diff --git a/testThreads.c b/testThreads.c
index b43cbd0..bff5391 100644
--- a/testThreads.c
+++ b/testThreads.c
@@ -105,10 +105,8 @@ main(void)
     for (repeat = 0;repeat < 500;repeat++) {
 	xmlLoadCatalog(catalog);
 
-	for (i = 0; i < num_threads; i++) {
-	    results[i] = NULL;
-	    tid[i] = (pthread_t) -1;
-	}
+        memset(results, 0, sizeof(*results)*num_threads);
+        memset(tid, 0xff, sizeof(*tid)*num_threads);
 
 	for (i = 0; i < num_threads; i++) {
 	    ret = pthread_create(&tid[i], NULL, thread_specific_data,
-- 
1.7.8.5

